File: android/src/main/kotlin/noise/protocol/Role.kt
package noise.protocol

/**
 * Defines the role of a participant in a Noise Protocol handshake.
 *
 * In every Noise handshake exactly two parties are involved: one that
 * initiates the connection and one that responds. The role determines
 * message ordering and which DH operations are performed at each step.
 *
 * @see HandshakeState
 * @see NoiseSession
 */
enum class Role {
    /** The party that sends the first handshake message. */
    INITIATOR,

    /** The party that receives the first handshake message and replies. */
    RESPONDER
}

File: ios/Sources/NoiseProtocol/Role.swift
import Foundation

/// Represents the role a party plays in a Noise Protocol handshake.
///
/// Each Noise handshake has exactly two participants: an initiator who sends the
/// first message and a responder who receives it.
public enum Role: Sendable {
    /// The party that initiates the handshake by sending the first message.
    case initiator
    /// The party that responds to the initiator's first message.
    case responder
}
